testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java

```diff
@@ -845,15 +845,15 @@ protected SqlOperatorFixture fixture() {
     final SqlOperatorFixture f = fixture();
     f.setFor(SqlStdOperatorTable.CAST, VmName.EXPAND);
 
-    f.checkScalarExact("cast( cast(1.25 as double) as integer)", 1);
-    f.checkScalarExact("cast( cast(-1.25 as double) as integer)", -1);
+    f.checkFails("cast( cast(1.25 as double) as integer)", "Value out of range.", true);
+    f.checkFails("cast( cast(-1.25 as double) as integer)", "Value out of range.", true);
     if (!f.brokenTestsEnabled()) {
       return;
     }
-    f.checkScalarExact("cast( cast(1.75 as double) as integer)", 2);
-    f.checkScalarExact("cast( cast(-1.75 as double) as integer)", -2);
-    f.checkScalarExact("cast( cast(1.5 as double) as integer)", 2);
-    f.checkScalarExact("cast( cast(-1.5 as double) as integer)", -2);
+    f.checkFails("cast( cast(1.75 as double) as integer)", "Value out of range.", true);
+    f.checkFails("cast( cast(-1.75 as double) as integer)", "Value out of range.", true);
+    f.checkFails("cast( cast(1.5 as double) as integer)", "Value out of range.", true);
```

Comment:
Besides, we could use OUT_OF_RANGE_MESSAGE instead of "Value out of range."

Reply:
nice suggestion, revised.